zebra-chain/src/amount.rs

```diff
@@ -786,24 +798,26 @@ mod test {
 
         // above max of i64 error
         let times = i64::MAX / MAX_MONEY;
-        let mut amounts = vec![MAX_MONEY.try_into()?];
+        let mut amounts: Vec<Amount<NonNegative>> = vec![MAX_MONEY.try_into()?];
```

Comment:
This was a tricky bit: Type inference started failing when I added the extra impl, but the compiler hints were a bit long and confusing.

Adding a known type usually helps, until you find a type error, or the compiler has enough types to continue.

(Also I changed to using `NonNegative` here.)